truss/templates/server/common/termination_handler_middleware.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+import signal
+from typing import Callable
+
+from fastapi import Request
+
+DEFAULT_TERM_DELAY_SECS = 5.0
+
+
+class TerminationHandlerMiddleware:
+    def __init__(
+        self,
+        on_stop: Callable[[], None],
+        on_term: Callable[[], None],
+        termination_delay_secs: float = DEFAULT_TERM_DELAY_SECS,
+    ):
+        self._outstanding_request_count = 0
+        self._on_stop = on_stop
```

Comment:
this might be a stupid question but what is the difference between stop and termination? Does it warrant a comment here?